### DISCUSSION TITLE
Welcome to Erasure Discussions!

### DISCUSSION THREAD

**Discussion by ian-campbell** [ian-campbell/Erasure#7] [General]
## 👋 Welcome!

Thank you for using Erasure and helping us make it better.

**Comment by cmaggiulli** [comment]
Is it normal for the DOM to not reflect the deletion of the comments?  I see the console log rolling and I see xhr requests being fired but I do not see anything reflected in the DOM. When I go to my activity page from different devices it appears as though no comments were removed?  Are these actions asynchronous? Do I need to wait for a period of time?

**Reply by ian-campbell** [reply]
The DOM should show the comments being deleted, roughly one every 5 seconds. It's not working for you? I just tested it again and it works for me.

**Comment by MECMwithShawn** [comment]
// === CONFIGURATION ===
const DELAY = 3000; // 3 second between deletions
const PAUSE = 1000; // 1 second between retries

let stopErasure = false; // Manual stop flag

// === UTILITIES ===
function autoScroll() {
  window.scrollTo({ left: 0, top: document.body.scrollHeight, behavior: "smooth" });
}

function getCommentElements() {
  return document.getElementsByClassName("YxbmAc");
}

function hasComments() {
  const comments = getCommentElements();
  if (comments.length > 0) {
    console.log(`erasure: ${comments.length} comment(s) available.`);
    return true;
  }
  return false;
}

function deleteComment(element, onComplete) {
  try {
    const deleteButtons = element.querySelectorAll(".VfPpkd-rymPhb-pZXsl");
    if (deleteButtons.length > 1) {
      deleteButtons[1].click();
    } else {
      throw new Error(); // fallback
    }
  } catch {
    const fallbackButtons = element.querySelectorAll(".VfPpkd-Bz112c-LgbsSe");
    if (fallbackButtons.length > 0) {
      fallbackButtons[0].click();
    } else {
      console.error("erasure: no delete button found for comment:", element);
    }
  }

  setTimeout(onComplete, DELAY);
}

// === MAIN LOOP ===
function processComments(index = 0) {
  if (stopErasure) {
    console.log("erasure: manually stopped.");
    return;
  }

  const comments = getCommentElements();

  if (comments.length > index) {
    deleteComment(comments[index], () => processComments(index));
  } else if (hasComments()) {
    console.log(`erasure: retrying after pause (${PAUSE}ms)...`);
    autoScroll();
    setTimeout(() => processComments(0), PAUSE);
  } else {
    console.log("erasure: no comments left to delete. Exiting.");
  }
}

// === START ===
processComments();
